packages/auth-types/index.d.ts

```diff
@@ -181,10 +183,16 @@ export interface IdTokenResult {
   };
 }
 
+export interface OAuthProviderOptions {
+  idToken?: string;
+  accessToken?: string;
+  rawNonce?: string;
+}
+
 export class OAuthProvider implements AuthProvider {
   providerId: string;
   addScope(scope: string): AuthProvider;
-  credential(idToken?: string, accessToken?: string): OAuthCredential;
+  credential(options: OAuthProviderOptions): OAuthCredential;
```

Comment:
It can also take a ID token string as I read from the comment. Should provide an overload signature for it.